framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -592,264 +523,346 @@ public void standardizeVariable(AnnotatedTypeMirror type, Element elt) {
                     return;
                 }
 
-                standardizeVariable(declarationTree, type, elt);
+                atVariableDeclaration(declarationTree, type, (VariableElement) elt);
                 return;
 
             default:
-                // It's not a variable (it might be METHOD, CONSTRUCTOR, CLASS, or INTERFACE, for
-                // example), so there is nothing to do.
+                // It's not a local variable (it might be METHOD, CONSTRUCTOR, CLASS, or INTERFACE,
+                // for example), so there is nothing to do.
                 break;
         }
     }
 
     /**
-     * Parse and standardize the expressions in dependent types in {@code type} as if they were
-     * written at {@code localVarPath}. For example, {@code @KeyFor("field") String} is changed to
-     * {@code @KeyFor("this.field")}
+     * Viewpoint-adapt all dependent type annotations to the method declaration, {@code
+     * methodDeclTree}. This method changes occurrences of formal parameter names to the "#2"
+     * syntax, and it removes expressions that contain other local variables.
      *
-     * @param localVarPath the expression is parsed as if it were written at this location
-     * @param type the type to parse; is side-effected by this method
+     * <p>If a Java expression in {@code atm} references local variables (other than formal
+     * parameters), the expression is removed from the annotation. This could result in dependent
+     * type annotations with empty lists of expressions. If this is a problem, subclasses can
+     * override {@link #buildAnnotation(AnnotationMirror, Map)} to do something besides creating an
+     * annotation with a empty list.
+     *
+     * @param methodDeclTree the method declaration to which the annotations are viewpoint-adapted
+     * @param atm type to viewpoint-adapt; is side-effected by this method
      */
-    private void parseToPath(TreePath localVarPath, AnnotatedTypeMirror type) {
-        Tree enclosingClass = TreePathUtil.enclosingClass(localVarPath);
-        TypeMirror enclosingType = TreeUtils.typeOf(enclosingClass);
-
-        JavaExpression receiver = JavaExpression.getPseudoReceiver(localVarPath, enclosingType);
-
-        JavaExpressionContext localContext =
-                new JavaExpressionContext(
-                        receiver,
-                        JavaExpression.getParametersOfEnclosingMethod(localVarPath),
-                        factory.getChecker());
-        standardizeAtm(localContext, localVarPath, type, /*removeErroneousExpressions=*/ false);
+    public void delocalize(MethodTree methodDeclTree, AnnotatedTypeMirror atm) {
```

Comment:
>Why isn't this called "atMethodDeclaration"? Is it because it "removes expressions that contain other local variables" rather than turning them into errors?

Yes, also, `StringToJavaExpression#atMethodDecl` does not remove local variables, so rename `delocalize` to `atMethodDecl` would be confusing.

It seems different to me than the other `at` methods because it is removing expressions.  Maybe `fromMethodBodyToMethodDecl`?